barretenberg/cpp/src/barretenberg/translator_vm_recursion/translator_recursive_verifier.cpp

```diff
@@ -0,0 +1,167 @@
+        batching_challenge_v_power = batching_challenge_v_power * batching_challenge_v;
+    }
+
+    relation_parameters.accumulated_result = compute_four_limbs(accumulated_result);
+};
+
+/**
+ * @brief This function verifies an TranslatorFlavor Honk proof for given program settings.
+ */
+template <typename Flavor>
+std::array<typename Flavor::GroupElement, 2> TranslatorRecursiveVerifier_<Flavor>::verify_proof(const HonkProof& proof)
+{
+    using Sumcheck = ::bb::SumcheckVerifier<Flavor>;
+    using PCS = typename Flavor::PCS;
+    using ZeroMorph = ::bb::ZeroMorphVerifier_<PCS>;
+    using VerifierCommitments = typename Flavor::VerifierCommitments;
+    using CommitmentLabels = typename Flavor::CommitmentLabels;
+    using Transcript = typename Flavor::Transcript;
+
+    StdlibProof<Builder> stdlib_proof = bb::convert_proof_to_witness(builder, proof);
+    transcript = std::make_shared<Transcript>(stdlib_proof);
+
+    // TODO(github.com/AztecProtocol/barretenberg/issues/985): Normally, the ECCVM verifier would have run
+    // before the translator and there will already by data in the transcript that can be hash to get the batching
+    // challenge. Once this is implemented the hack can be removed.
+    transcript->template receive_from_prover<BF>("init");
+    batching_challenge_v = transcript->template get_challenge<BF>("Translation:batching_challenge");
+
+    VerifierCommitments commitments{ key };
+    CommitmentLabels commitment_labels;
+
+    // ? BF
+    const auto circuit_size = transcript->template receive_from_prover<FF>("circuit_size");
+    ASSERT(static_cast<uint32_t>(circuit_size.get_value()) == key->circuit_size);
+    evaluation_input_x = transcript->template receive_from_prover<BF>("evaluation_input_x");
+
+    const BF accumulated_result = transcript->template receive_from_prover<BF>("accumulated_result");
+
+    put_translation_data_in_relation_parameters(evaluation_input_x, batching_challenge_v, accumulated_result);
+
+    // Get commitments to wires and the ordered range constraints that do not require additional challenges
+    for (auto [comm, label] : zip_view(commitments.get_wires_and_ordered_range_constraints(),
+                                       commitment_labels.get_wires_and_ordered_range_constraints())) {
+        comm = transcript->template receive_from_prover<Commitment>(label);
+    }
+
+    // Get permutation challenges
+    FF gamma = transcript->template get_challenge<FF>("gamma");
+
+    relation_parameters.beta = 0;
+    relation_parameters.gamma = gamma;
+    relation_parameters.public_input_delta = 0;
+    relation_parameters.lookup_grand_product_delta = 0;
+
+    // Get commitment to permutation and lookup grand products
+    commitments.z_perm = transcript->template receive_from_prover<Commitment>(commitment_labels.z_perm);
+
+    // Execute Sumcheck Verifier
+    const size_t log_circuit_size = numeric::get_msb(static_cast<uint32_t>(circuit_size.get_value()));
```

Comment:
Please create an issue to fix this parameter in the future. (We'll need 1 size and it can't be defined by the transcript. Your circuit here depends on the transcript)

Reply:
Opened issue

Reply:
discussed offline, established not an issue as long as we have recursive verifiers for variable circuit sizes